client/blocks/eligibility-warnings/excessive-disk-space.tsx

```diff
@@ -0,0 +1,41 @@
+import { isBusinessPlan, isEcommercePlan, isProPlan } from '@automattic/calypso-products';
+import { localize, LocalizeProps } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import PlanStorage from 'calypso/blocks/plan-storage';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import { getSiteSlug, getSitePlanSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const ExcessiveDiskSpace = ( { translate }: { translate: LocalizeProps[ 'translate' ] } ) => {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, selectedSiteId ) );
+	const canUserUpgrade = useSelector( ( state ) =>
+		canCurrentUser( state, selectedSiteId ?? 0, 'manage_options' )
+	);
+	const sitePlanSlug = useSelector( ( state ) => getSitePlanSlug( state, selectedSiteId ) ?? '' );
+
+	const planHasTopStorageSpace =
+		isBusinessPlan( sitePlanSlug ) || isEcommercePlan( sitePlanSlug ) || isProPlan( sitePlanSlug );
+	const displayUpgradeLink = canUserUpgrade && ! planHasTopStorageSpace;
+
+	return (
+		<div>
+			{ translate( "Your site's disk usage exceeds the amount provided by its plan." ) }
```

Comment:
> Not sure how to elegantly word this?

Maybe something like "Your site does not have enough free disk space." That's true for 97% or 110%.

Reply:
Also, if we finally opt for the more generic "Your site does not have enough free disk space." message, we'd need to change the "Meet disk quote" heading to something else as well – maybe "Free up storage"?

Reply:
Oh, another thought on the wording suggestion: We should probably call it "storage space" not "disk space". I don't think we use the term "disk space" in marketing or support docs.

Reply:
Updated `"Your site's disk usage exceeds the amount provided by its plan."` -> `Your site does not have enough free storage space.`

For "Free up storage", I'm concerned that it doesn't mention buying more storage, since that's an easier option that brings in additional revenue, and also doesn't encourage the user to do something they might regret (delete important files without understanding the implications, and contact us for support later). I tried `'Purchase or free up storage'` - any thoughts or suggestions?

Reply:
Maybe "Increase storage space"? Another thought, maybe we should use "available" instead of "free" so as not to confuse it with the "free plan."

So the message would read:

> To activate hosting you'll need to:
> - Increase storage space
> Your site does not have enough available storage space.
> Please purchase a plan with additional storage or contact our support team for help.

Reply:
Rebased and updated wording